api/v1alpha1/accesslogging_types.go

```diff
@@ -30,8 +30,29 @@ type ProxyAccessLogSetting struct {
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=50
 	Sinks []ProxyAccessLogSink `json:"sinks"`
+	// Type defines the component emitting the accesslog, such as Listener and Route.
+	// If type not defined, the setting would apply to all accesslog types supported by envoy gateway.
+	// If type is defined, the default envoy gateway accesslog configuration for the mentioned component would be
```

Comment:
for my understanding this implies that a `nil` ProxyAccessLogSetting is different that a ProxyAccessLogSetting with a `nil` type set ?

Reply:
Right. Let's simplify this and make sure that we don't have surprising behavior for end users where unscoped logs suddenly disappear. 
- if type is not set - behavior remains as today (applied to routes as-is, to listener with additional NR filter)
- if type is set - applies to relevant component.